apps/election-manager/src/components/ExportFinalResultsModal.test.tsx

```diff
@@ -0,0 +1,201 @@
+  expect(fetchMock.called('/convert/results/process')).toBe(true)
+  expect(fetchMock.called('/convert/results/output?name=name')).toBe(true)
+  expect(fetchMock.called('/convert/reset')).toBe(true)
+
+  fireEvent.click(getByText('Close'))
+  expect(closeFn).toHaveBeenCalled()
+})
+
+test('render export modal with errors when appropriate and clears errors when closed', async () => {
+  const mockKiosk = fakeKiosk()
+  window.kiosk = mockKiosk
+
+  // When inputFiles doesn't return 2 expected files the saving code will error.
+  fetchMock.getOnce('/convert/results/files', {
+    inputFiles: [],
+    outputFiles: [],
+  })
+
+  const closeFn = jest.fn()
+  const { getByText } = renderInAppContext(
+    <ExportFinalResultsModal isOpen onClose={closeFn} />,
+    {
+      usbDriveStatus: UsbDriveStatus.mounted,
+    }
+  )
+  getByText('Export Results')
+
+  fireEvent.click(getByText('Export'))
+  await waitFor(() => getByText(/Download Failed/))
+  getByText(/Failed to save results./)
+  getByText(/Cannot read property 'name' of undefined/)
+
+  fireEvent.click(getByText('Close'))
+  expect(closeFn).toHaveBeenCalled()
+  getByText('Export Results') // Closing should reset back to the starting export screen
+})
+
+test('when exporting results to a mounted usb changing the filename works as expected', async () => {
+  // Changing alpha numeric works, captial lowercase
+  // hyphens and underscores work
+  // and percands, dots, asterisk dont work
+  // When saving the name is actually what is saved.
+  const mockKiosk = fakeKiosk()
+  window.kiosk = mockKiosk
+  mockKiosk.getUsbDrives.mockResolvedValue([fakeUsbDrive()])
+
+  fetchMock.getOnce('/convert/results/files', {
+    inputFiles: [{ name: 'name' }, { name: 'name' }],
+    outputFiles: [{ name: 'name' }],
+  })
+
+  fetchMock.post('/convert/results/submitfile', { body: { status: 'ok' } })
+  fetchMock.post('/convert/results/process', { body: { status: 'ok' } })
+
+  fetchMock.getOnce('/convert/results/output?name=name', { body: '' })
+
+  fetchMock.post('/convert/reset', { body: { status: 'ok' } })
+
+  const closeFn = jest.fn()
+  const { getByText, getByTestId } = renderInAppContext(
+    <ExportFinalResultsModal isOpen onClose={closeFn} />,
+    {
+      usbDriveStatus: UsbDriveStatus.mounted,
+    }
+  )
+  getByText('Export Results')
+  const filename = getByTestId('filename-input').closest('input')!
+  expect(filename.value).toBe('votingworks-live-results_2020-03-14_01-59-26')
+
+  // Test you can change the filename to basic alphanumeric names with legal characters
+  fireEvent.change(filename, { target: { value: 'test-filename' } })
+  expect(filename.value).toBe('test-filename')
+  fireEvent.change(filename, { target: { value: 'Test-fileName1__' } })
+  expect(filename.value).toBe('Test-fileName1__')
+
+  // Illegal characters are not allowed to be inputted into the filename.
+  fireEvent.change(filename, { target: { value: 'test-filename.Hello' } })
```

Comment:
Periods are okay, really. You could maybe argue they're confusing, but they're not "illegal" in any fs IO way.